IPython/core/tests/test_display.py

```diff
@@ -195,6 +195,14 @@ def test_displayobject_repr():
     j._show_mem_addr = False
     nt.assert_equal(repr(j), '<IPython.core.display.Javascript object>')
 
+@mock.patch('warnings.warn')
+def test_encourage_iframe_over_html(m_warn):
+    display.HTML('<br />')
+    m_warn.assert_not_called()
```

Comment:
Ah I would have use [`assert_warns`](https://docs.pytest.org/en/latest/warnings.html#warns), but that looks good !

Reply:
Didn’t know about that one, thanks!

Reply:
I had a look at the link that you posted, it seems specific to pytest, which IPython doesn't use. Did you mean to send [this link](https://docs.python.org/3/library/warnings.html#testing-warnings)? Let me know if I misunderstood something.

Reply:
Ah, yes, sorry, and [this](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertWarns) I just get confused between test runners.